test-requirements.txt

```diff
@@ -3,7 +3,7 @@ coverage[toml]
 flake8-docstrings
 flake8-builtins
 isort
-juju==3.2.0.0
+git+https://github.com/juju/python-libjuju
```

Comment:
Why this change?

Reply:
Juju 3.2.0.0 is currently broken with a pinned version of PyYAML not higher than 6.0. This fixes it because the fix is merged, but there is no release yet. I know there is work on a release happening now, I will update this PR once it is released.

Reply:
The fix has been merged and should be published soon, so I updated the dependency to the new version. CI will fail until it is published to PyPi.